simulations/vip-306/simulations.ts

```diff
@@ -0,0 +1,49 @@
+import { TransactionResponse } from "@ethersproject/providers";
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { expectEvents } from "../../src/utils";
+import { forking, testVip } from "../../src/vip-framework";
+import vip303, { BNB, BTC, RESILIENT_ORACLE } from "../../vips/vip-306/bscmainnet";
+import BOUND_VALIDATOR_ABI from "./abi/boundValidator.json";
+import REDSTONE_ORACLE_ABI from "./abi/redstoneOracle.json";
+import RESILIENT_ORACLE_ABI from "./abi/resilientOracle.json";
+
+forking(38912197, () => {
+  const provider = ethers.provider;
+  let oracle: Contract;
+
+  before(async () => {
+    oracle = new ethers.Contract(RESILIENT_ORACLE, RESILIENT_ORACLE_ABI, provider);
+  });
+
+  describe("Pre-VIP behavior", async () => {
+    it("validate price", async () => {
+      const bnbPrice = await oracle.getPrice(BNB);
+      expect(bnbPrice).to.equal(parseUnits("620.30", 18));
+
+      const btcPrice = await oracle.getPrice(BTC);
+      expect(btcPrice).to.equal(parseUnits("70979.20075", 18));
+    });
+  });
+
+  testVip("Update BTC and BNB Price Config", vip303(60 * 60 * 24 * 7), {
+    callbackAfterExecution: async (txResponse: TransactionResponse) => {
+      await expectEvents(txResponse, [REDSTONE_ORACLE_ABI], ["TokenConfigAdded"], [4]);
+      await expectEvents(txResponse, [BOUND_VALIDATOR_ABI], ["ValidateConfigAdded"], [2]);
+      await expectEvents(txResponse, [RESILIENT_ORACLE_ABI], ["TokenConfigAdded"], [2]);
+    },
+  });
+
+  describe("Post-VIP state", () => {
+    it("validate price", async () => {
```

Comment:
I'd feel safer if we had failure simulations here as well, e.g. RedStone and Chainlink reporting different prices. BNB and BTC are quite huge markets, and we'd rather not have any "invalid price" errors there, especially on a volatile market when we need liquidations to happen. Wdyt?

Reply:
We have this: https://github.com/VenusProtocol/oracle/pull/194